File: examples/mortar-app/src/main/java/architect/examples/mortar_app/Parceler.java
package architect.examples.mortar_app;

import android.os.Parcelable;

import org.parceler.Parcels;

import architect.Screen;
import architect.ScreenParceler;

/**
 * PathParceler that uses Parceler library
 *
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
public class Parceler implements ScreenParceler {

    @Override
    public Parcelable wrap(Screen path) {
        return Parcels.wrap(path);
    }

    @Override
    public Screen unwrap(Parcelable parcelable) {
        return Parcels.unwrap(parcelable);
    }
}

File: examples/kotlin_app/src/main/kotlin/architect/kotlinapp/Parceler.kt
package architect.kotlinapp

import android.os.Parcelable
import architect.ScreenParceler
import architect.Screen
import org.parceler.Parcels

/**
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
class Parceler : ScreenParceler<Screen> {

    override fun wrap(path: Screen?): Parcelable? {
        return Parcels.wrap<Screen>(path)
    }

    override fun unwrap(parcelable: Parcelable?): Screen? {
        return Parcels.unwrap<Screen>(parcelable)
    }
}